src/client/datascience/kernel-launcher/kernelEnvVarsService.ts

```diff
@@ -48,22 +48,23 @@ export class KernelEnvironmentVariablesService {
                 });
         }
 
-        if (interpreter?.envType !== EnvironmentType.Conda) {
```

Comment:
This was the problem, when dealing with non-conda we'd not load the env variables.

Reply:
Could easily repro on linux as well.